deploy/overlays/namespace_scoped/deployment.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: grafana-operator-controller-manager
+spec:
+  template:
+    spec:
+      containers:
+        - name: manager
+          args:
+            - --health-probe-bind-address=:8081
+            - --metrics-bind-address=127.0.0.1:8080
+            - --leader-elect
+            - --namespace=true
```

Comment:
@pb82 I pushed a commit with the patches usage. But I saw your comment from earlier.
I thought the usage was just `--namespace=true` but reading your comment it needs to be `--namespace=grafana-operator` or which ever namespace you want to monitor?

Is there a reason for this? Can't we assume that if the user wants to only deploy to one namespace we can also assume that he wants the grafana instance to be together with the operator.
Else you still need to write a clusterrole for the operator, sure you would still only do a rolebinding but still.

From the controller I assume it's rather easy to check witch namespace it's deployed to.

Reply:
If we want to keep this to a similar way we could use the downward API, but then we would have to change over to a env variable.

env:
    - name: namespace
      valueFrom:
        fieldRef:
          fieldPath: metadata.namespace

Reply:
True, if we always assume that `--namespace` means the Operator namespace, we wouldn't have to worry about injecting the namespace into the args somehow (and could also just use a patch instead of a replacement).